File: compat/python-psycopg/smoke.py
import os

import psycopg


def main() -> None:
    url = os.environ.get(
        "DATABASE_URL",
        "postgres://postgres:postgres@127.0.0.1:58432/pgkinetic",
    )
    with psycopg.connect(url) as conn:
        with conn.cursor() as cursor:
            cursor.execute(
                "select balance_cents from accounts where email = %s",
                ("alice@example.com",),
            )
            row = cursor.fetchone()

    if row is None:
        raise RuntimeError("expected one row")
    if row[0] != 1000:
        raise RuntimeError(f"expected balance 1000, got {row[0]}")

    print("python psycopg smoke passed")


if __name__ == "__main__":
    main()

File: compat/node-pg/smoke.js
import pg from "pg";

const connectionString =
  process.env.DATABASE_URL ??
  "postgres://postgres:postgres@127.0.0.1:58432/pgkinetic";

const client = new pg.Client({ connectionString });

await client.connect();
const result = await client.query({
  name: "account-balance",
  text: "select balance_cents from accounts where email = $1",
  values: ["alice@example.com"],
});
await client.end();

const balance = Number(result.rows[0].balance_cents);
if (balance !== 1000) {
  throw new Error(`expected balance 1000, got ${balance}`);
}

console.log("node pg smoke passed");
